src/codegen/riscv64/macro-assembler-riscv64.cc

```diff
@@ -3630,32 +3630,26 @@ void TurboAssembler::DsubOverflow(Register dst, Register left,
 }
 
 void TurboAssembler::MulOverflow(Register dst, Register left,
-                                 const Operand& right, Register overflow) {
+                                 const Operand& right, Register overflow,
+                                 Register scratch) {
   BlockTrampolinePoolScope block_trampoline_pool(this);
   Register right_reg = no_reg;
-  Register scratch = t5;
```

Comment:
Why pass in the scratch register instead of using this technique to get one? I believe the common practice is to only pass in a scratch register if its value needs to be live-out. Otherwise, we can use the available techniques to use the 3 available assembler temporary registers.

Reply:
Do you mean wo can using t3 、t5 、t6 as assembler temporary registers? If so , i agree you to using it.

Reply:
Yes, you can use them, with the correct setup. Here, the line:
```
BlockTrampolinePoolScope block_trampoline_pool(this);
```
allows you to safely use `t5` and `t6`.